src/cli/guess.rs

```diff
@@ -18,22 +17,24 @@ pub enum Error {
     ReadXdrNext(#[from] crate::next::Error),
     #[error("error reading file: {0}")]
     ReadFile(#[from] std::io::Error),
+    #[error("couldn't guess XDR type")]
+    InvalidType,
 }
 
 #[derive(Args, Debug, Clone)]
 #[command()]
 pub struct Cmd {
-    /// File to decode, or stdin if omitted
+    /// XDR or files containing XDR to decode, or stdin if empty
     #[arg()]
-    pub file: Option<PathBuf>,
+    pub input: Vec<OsString>,
```

Comment:
Afaict there is a change in behaviour here as part of this refactor. The input was changed from being one path, to be a list of paths or inputs. This command's API is intended to return only a list of possible types for that one value. If it returns a list of possible types for multiple inputs, the output is going to be pretty confusing.

Unclear on the motivation of the refactor here, the guess command doesn't need to use parse_input, it's `file` function is already very simple. We don't need the code to be super DRY as it isn't material debt.